### PR TITLE
Update operating mode upon network disagreement.

### PR DESCRIPTION
**Pull request by mtrippled**

### CODE REVIEW

**Comment by nbougalis** [comment]
@mtrippled please rebase this

File: src/ripple/app/misc/NetworkOPs.cpp [hunk 1]
```diff
@@ -620,11 +623,16 @@ static std::array<char const*, 5> const stateNames {{
     "full"}};
 
 #ifndef __INTELLISENSE__
-static_assert (NetworkOPs::omDISCONNECTED == 0, "");
-static_assert (NetworkOPs::omCONNECTED == 1, "");
-static_assert (NetworkOPs::omSYNCING == 2, "");
-static_assert (NetworkOPs::omTRACKING == 3, "");
-static_assert (NetworkOPs::omFULL == 4, "");
+static_assert (static_cast<std::size_t>(
```
[the remark this answers is not in the document]
Comment (mtrippled) [reply src/ripple/app/misc/NetworkOPs.cpp]: fixed

File: src/ripple/app/misc/NetworkOPs.h [hunk 2]
```diff
@@ -76,7 +76,7 @@ class NetworkOPs
 public:
     using clock_type = beast::abstract_clock <std::chrono::steady_clock>;
 
-    enum OperatingMode
+    enum class OperatingMode
```
[the remark this answers is not in the document]
Comment (mtrippled) [reply src/ripple/app/misc/NetworkOPs.h]: fixed

**Comment by mtrippled** [comment]
In addition to the review comments, I fixed a race condition that I introduced in the original commit. There was a brief window in which the OperatingMode causing online_delete to abend could have changed by the time the log message was written.

**Pull request closed without merging**